### PR TITLE
add elastic apm plugins

### PR DESCRIPTION
**Pull request by timogoebel**
This adds [foreman_supervisory_authority](https://github.com/timogoebel/foreman_supervisory_authority) ~~and [smart_proxy_elasticapm](https://github.com/timogoebel/smart_proxy_elasticapm)~~.

The elastic apm agent requires at last ruby 2.3.0, so we won't be able to run it on smart-proxy. :-(

### CODE REVIEW

**Comment by timogoebel** [comment]
[test rpm]

File: packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -pa .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/.gitignore
+%{gem_instdir}/.hound.yml
```
Comment (ekohl) [inline packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec]:
```suggestion
%exclude %{gem_instdir}/.hound.yml
```

File: packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+%exclude %{gem_instdir}/.rubocop.yml
+%{gem_instdir}/CODE_OF_CONDUCT.md
```
Comment (ekohl) [inline packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec]: Maybe this should be part of `%files doc`?

```suggestion
%doc %{gem_instdir}/CODE_OF_CONDUCT.md
```

File: packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+%{gem_instdir}/Jenkinsfile
```
Comment (ekohl) [inline packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec]:
```suggestion
%exclude %{gem_instdir}/Jenkinsfile
```

File: packages/plugins/rubygem-http/rubygem-http.spec [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        %{buildroot}%{gem_dir}/
+
+%files
+%dir %{gem_instdir}
+%{gem_instdir}/.coveralls.yml
+%exclude %{gem_instdir}/.gitignore
+%exclude %{gem_instdir}/.rubocop.yml
+%exclude %{gem_instdir}/.travis.yml
+%exclude %{gem_instdir}/.yardopts
+%{gem_instdir}/CHANGES.md
```
Comment (ekohl) [inline packages/plugins/rubygem-http/rubygem-http.spec]: Maybe part of `%files doc`

```suggestion
%doc %{gem_instdir}/CHANGES.md
```

File: packages/plugins/rubygem-http/rubygem-http.spec [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+%{?scl:EOF}
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -pa .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%files
+%dir %{gem_instdir}
+%{gem_instdir}/.coveralls.yml
```
Comment (ekohl) [inline packages/plugins/rubygem-http/rubygem-http.spec]:
```suggestion
%exclude %{gem_instdir}/.coveralls.yml
```

File: packages/plugins/rubygem-http/rubygem-http.spec [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+%exclude %{gem_instdir}/.gitignore
+%exclude %{gem_instdir}/.rubocop.yml
+%exclude %{gem_instdir}/.travis.yml
+%exclude %{gem_instdir}/.yardopts
+%{gem_instdir}/CHANGES.md
+%{gem_instdir}/Guardfile
```
Comment (ekohl) [inline packages/plugins/rubygem-http/rubygem-http.spec]:
```suggestion
%exclude %{gem_instdir}/Guardfile
```

File: packages/plugins/rubygem-http_parser.rb/rubygem-http_parser.rb.spec [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+%exclude %{gem_instdir}/http_parser.rb.gemspec
+%{gem_libdir}
+%{gem_instdir}/tasks
+%exclude %{gem_cache}
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%{gem_instdir}/Gemfile
+%{gem_instdir}/Gemfile.lock
```
Comment (ekohl) [inline packages/plugins/rubygem-http_parser.rb/rubygem-http_parser.rb.spec]:
```suggestion
%exclude %{gem_instdir}/Gemfile.lock
```

**Comment by timogoebel** [comment]
@ekohl: I've released the plugin and added packaging here. Mind taking a look?

**Comment by ekohl** [comment]
[test rpm]

**Comment by timogoebel** [comment]
[test rpm]

File: packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec [hunk 8]
```diff
@@ -0,0 +1,98 @@
+%{?scl:%scl_package rubygem-%{gem_name}}
```
Comment (evgeni) [inline packages/plugins/rubygem-elastic-apm/rubygem-elastic-apm.spec]: I guess this has been generated from the `scl` template, but the file is missing the `# template: scl` header, could you add it (back) please to make gem updates easier?

**Comment by evgeni** [comment]
```
BUILDSTDERR: error: Installed (but unpackaged) file(s) found:
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/.jenkins_exclude.yml
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/.jenkins_framework.yml
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/.jenkins_ruby.yml
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/Jenkinsfile
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/bin/check_paths_for_matches.py
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/downstreamTests.groovy
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/jobs/apm-agent-ruby-downstream.yml
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/jobs/apm-agent-ruby-mbp.yml
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/jobs/apm-agent-ruby.yml
BUILDSTDERR:    /opt/theforeman/tfm/root/usr/share/gems/gems/elastic-apm-2.9.1/.ci/jobs/defaults.yml
```

I think you need some more `%exclude`s

**Comment by timogoebel** [comment]
> I think you need some more `%exclude`s

Re-generated the spec file.

**Comment by timogoebel** [comment]
@ekohl: Anything missing here?

**Pull request merged**